t3374: Declare do_update before setup tool update prompt

## What

Fix the pre-existing ShellCheck SC2154 warning in `setup-modules/plugins.sh` where `do_update` is referenced after `setup_prompt do_update ...` without a prior local declaration.

## Why

The release preflight for #22059 failed because `setup-modules/plugins.sh` changed in that PR and release preflight ShellCheck now checks changed files since the last release. The warning is pre-existing, but it blocks release whenever the file is included in a patch release.

## Files to modify

- EDIT: `setup-modules/plugins.sh:660-710` — declare `local do_update=""` before calling `setup_prompt do_update ...`.

## Reference pattern

Follow existing local variable declarations in nearby setup helper functions.

## Verification

- `shellcheck setup-modules/plugins.sh` or, if unrelated legacy warnings remain, verify SC2154 is absent for `do_update`.
- `.agents/scripts/version-manager.sh release patch` preflight no longer fails on this warning.


## Brief Workflow

This issue body is composed under `.agents/workflows/brief.md`. Before implementation, ensure it contains files to modify, a reference pattern, verification commands, and concrete acceptance criteria.

<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.13.51 plugin for [OpenCode](https://opencode.ai) v1.14.30 with gpt-5.5 spent 1h 7m and 299,350 tokens on this as a headless worker.

--- a/setup-modules/plugins.sh
+++ b/setup-modules/plugins.sh
@@ -699,6 +699,7 @@ check_tool_updates() {
 	bash "$tool_check_script" --quiet
 	echo ""
 
+	local do_update=""
 	setup_prompt do_update "Update all outdated tools now? [Y/n]: " "Y"
 
 	if [[ "$do_update" =~ ^[Yy]?$ || "$do_update" == "Y" ]]; then

